crates/system-manager/src/main.rs

```diff
@@ -185,10 +232,18 @@ fn go(args: Args) -> Result<()> {
     let Args {
         action,
         target_host,
-        use_remote_sudo,
+        legacy_use_remote_sudo,
         nix_options,
     } = args;
 
+    // Check for deprecated flag
+    if legacy_use_remote_sudo {
+        bail!(
+            "--use-remote-sudo has been removed. Use --sudo on the subcommand instead.\n\
+             Example: system-manager switch --sudo --flake ."
+        );
+    }
+
```

Comment:
Ah, if this is how we are handling deprecation we should probably just throw the error at the parsing level by not including the option in the command line parser.